content/post/practical-stegen.Rmd

```diff
@@ -0,0 +1,861 @@
+```
+
+</details>
+
+
+
+## Age and gender distribution of the cases
+
+We use the same principles as before to visualise the distribution of illness by
+age and gender. To split the plot into different panels according to `sex`, we
+use `facet_grid()` (see previous extra info on *ggplot2* customisation for
+further details otions):
+
+```{r stegen_case_age_sex}
+ggplot(stegen) + 
+  geom_histogram(aes(x = age, fill = ill)) +
+  scale_fill_manual("Illness", values = c("non case" = "#66cc99", "case" = "#990033")) +
+  facet_grid(sex ~ .) + 
+  labs(title = "Cases by age and gender") + 
+  theme_light()
+```
+
+Illness does not seem to be depending on age or gender. This can be tested: this
+is the topic of the next and final section.
+
+
+
+# Looking for the culprits
+
+## Univariate tests
+
+Methods for testing the association between two variables can be broken down in 3 types,
+depending on which types these variables are:
+
+1. **2 quantitative variables**: Pearson's correlation coefficient ($r$) and similar
+  methods
+2. **1 quantitative, 1 categorical**: ANOVA types of approaches; particular case with
+  2 groups: Student's $t$-test
+3. **2 categorical variables**: Chi-squared test on the 2x2 table
+  (a.k.a. *contingency* table) and similar methods (e.g. Fisher's exact test)
+
+
+We can use these approaches to test if the disease is linked to any of the other
+recorded variables. As illness itself is a categorical variable, only approaches
+of type 2 and 3 will be illustrated here.
+
+
+### Is illness linked to age?
+
+We can use the function `t.test()` to test if the average age is different across
+illness status. As this test assumes that the two categories exhibit similar
+variation, we first ensure that the variances are comparable using Bartlett's
+test. The syntax `variable ~ group` is used to indicate the variable of interest (left hand-side), and the group
+ (right hand-side):
+
+```{r stegen_bartlett_test}
+bartlett.test(stegen$age ~ stegen$ill)
+```
+
+The resulting p-value (`r round(bartlett.test(stegen$age ~ stegen$ill)[["p.value"]], 3)`) suggests the variances are indeed comparable. Wecan thus proceed to a $t$-test:
+
+
+```{r stegen_t_test}
+t.test(stegen$age ~ stegen$ill, var.equal = TRUE)
+```
+
+The p-value of `r round(t.test(stegen$age ~ stegen$ill, var.equal = TRUE)[["p.value"]], 3)` confirms the previous graphics: the illness does not seem to be linked to age.
+
+
+
+### Is illness linked to gender?
+
+To test the association between gender and illness (2 categorical variables), we
+first build a 2-by-2 (contingency) table, using:
+
+```{r stegen_2by2}
+tab_ill_sex <- table(stegen$ill, stegen$sex)
+tab_ill_sex
+```
+
+Note that proportions can be obtained using `prop.table`:
+
+```{r stegen_prop_table}
+## basic proportions
+prop.table(tab_ill_sex)
+
+## expressed in %, rounded:
+round(100 * prop.table(tab_ill_sex))
+```
+
+Once a contingency table has been built, the Chi-square test can be run using `chisq.test`:
```

Comment:
show after this how to pull the various parts in to one table? with counts prop and a p val?